tests/subsys/net/openthread/rpc/server/src/srp_client_suite.c

```diff
@@ -0,0 +1,826 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <mock_nrf_rpc_transport.h>
+#include <ot_rpc_ids.h>
+#include <test_rpc_env.h>
+
+#include <zephyr/net/net_ip.h>
+#include <zephyr/fff.h>
+#include <zephyr/ztest.h>
+
+#include <openthread/srp_client.h>
+
+#define TEST_PORT               55555
+#define TEST_TTL                120
+#define TEST_LEASE_INTERVAL     UINT32_MAX
+#define TEST_KEY_LEASE_INTERVAL 0xcafecafe
+#define TEST_PRIORITY           10
+#define TEST_WEIGHT             20
+
+#define TEST_HOSTNAME "hostname.local"
+#define CBOR_HOSTNAME 0x6E, 0x68, 0x6F, 0x73, 0x74, 0x6E, 0x61, 0x6D, 0x65, \
```

Comment:
defining test strings as 'c', 'h', 'a', 'r', 's' would be much less cryptic then using hex encoding